src/pages/index.tsx

```diff
@@ -162,9 +146,19 @@ export default function Home({
           optionFilterProp="children"
           onSelect={(_value, option) => setSelectedTrain(option.value)}
           onClear={() => setSelectedTrain(null)}
-          filterOption={(input, option) =>
-            (option?.label ?? "").toLowerCase().includes(input.toLowerCase())
-          }
+          filterOption={(input, option) => {
+            if (!option) return false;
+
+            const terms = [
+              option.label,
+              option.arrivalStationName,
+              option.arrivalStationShortCode,
+              option.departureStationName,
+              option.departureStationShortCode,
+            ];
+
+            return terms.some((term) => term.toLowerCase().includes(input.toLowerCase()));
```

Comment:
maybe also split input by spaces and search each one seperatedly so searching "oulu helsinki" would be possible

Reply:
Should it search for rows that contain Oulu AND Helsinki, or Oulu OR Helsinki?

Reply:
I'd prefer an AND search

Reply:
Done
![image](https://github.com/Chicken/VenaaRauhassa/assets/38920928/3d6ff609-2c9d-4973-9954-b63b985d740e)